homeassistant/components/bayesian/helpers.py

```diff
@@ -29,7 +29,12 @@ class Observation:
     below: float | None
     value_template: Template | None
     observed: bool | None = None
-    id: str = field(default_factory=lambda: str(uuid.uuid4()))
+    id: uuid.UUID = field(default_factory=uuid.uuid4)
+
+    def __post_init__(self) -> None:
+        """Observations will be keyed in dicts by their value_template OR entity_id."""
+
+        assert self.entity_id is not None or self.value_template is not None
```

Comment:
I don't think this should be in the code.
If you wish to add this, I think it is better as a comment or a docstring on `entity_id` and/or `value_template`